### PR TITLE
Ask log type at the begginng of using whylog

### PR DESCRIPTION
**Pull request by andrzejgorski**

### CODE REVIEW

File: whylog_vim/gui/__init__.py [hunk 1]
```diff
@@ -53,17 +52,23 @@ def get_button_name(self):
...
-        # TODO add proper host
-        host = 'localhost'
         cursor_position = VimUIWrapper.get_cursor_offset()
         line_content = VimUIWrapper.get_current_line()
-        line_source = LineSource(host, filename)
-        return FrontInput(cursor_position, line_content, line_source)
+        line_source = self.get_line_source()
+        front_input = FrontInput(cursor_position, line_content, line_source)
+        if self.log_type:
+            front_input.log_type = self.log_type
```
Comment (kgromadzki) [inline whylog_vim/gui/__init__.py]: FrontInput hasn't field log_type. In my opinion this if is unnecessary.. return FrontInput created in 68 lline.

File: whylog_vim/output_formater/input_windows_messages.py [hunk 2]
```diff
@@ -78,3 +79,30 @@ def get_edit_regex_message(cls, line_content, old_regex):
...
+            partial(read_function, log_type), (FunctionNames.READ_LOG_TYPE, log_type.name)
+        )
+        output.add(LogType.NAME % log_type.name)
+        output.create_button(*create_button_param)
+        for matcher in log_type.filename_matchers:
+            host_pattern = matcher.host_pattern or DefaultContent.UNDEFINED
+            path_pattern = matcher.path_pattern or DefaultContent.UNDEFINED
+            super_parser = matcher.super_parser.regex.pattern or DefaultContent.UNDEFINED
+
+            output.add(LogType.HOST_PATTERN % host_pattern)
```
Comment (kgromadzki) [inline whylog_vim/output_formater/input_windows_messages.py]: Here is assumption that LogType has one matcher. Acceptable in minimal version

Comment (andrzejgorski) [inline whylog_vim/output_formater/input_windows_messages.py]: In 92 line is a loop for all matchers.

File: whylog_vim/proxy/__init__.py [hunk 3]
```diff
@@ -14,8 +16,10 @@ def __init__(self, editor):
         self.log_reader = LogReaderProxy(log_reader, config, editor, self)
         self.teacher = TeacherProxy(self.teacher_generator(), config, self.editor, self)
         self._state = States.EDITOR_NORMAL
+        self.log_type = None
```
Comment (ppolewicz) [inline whylog_vim/proxy/__init__.py]: it looks like whylog proxy keeps one log type for just the file which has been opened, when the interface that we agreed to was supposed to keep a mapping from many files (which can be open by vim) which can be assigned to many logtypes. So this code would work only if no more than one non-anonymous file type is open. Please change it to an appropriate mapping.

**Pull request merged**